inspector/src/components/graph/canvasGraphService.ts

```diff
@@ -49,7 +51,7 @@ export class CanvasGraphService {
         // Keep only visible and non empty datasets and get a certain window of items.

```

Comment:
Could add in the todo as well to limit allocations ? to prevent extra GC if draw is called at high frequency

Reply:
Hmm I've added the comment but I'm having a bit of trouble right now imagining how I'd reduce the allocations happening due to the map.

The alternative I'm seeing right now is we slice on every iteration the items we need. But that feels just as bad. I may need to think of it more, but if you have any suggestions I'm open to them

Reply:
you could probably keep a data array in memory and manually filter from source into this array preventing recreation of the filter then map and also spread operator  but the code might be harder to read so it really depends of the draw frequency ?

Reply:
I have made note of this and will explore possible solutions including the ones we discussed in our offline chat (Potentially using typed arrays). Will continue thinking about how to reduce the garbage collection pressure from this filter and map.